oi/type_graph/NameGen.cpp

```diff
@@ -152,4 +152,28 @@ void NameGen::visit(Typedef& td) {
   accept(td.underlyingType());
 }
 
+void NameGen::visit(CycleBreaker& b) {
+  accept(b.to());
+
+  std::string name = "fake_" + b.to().name();
```

Comment:
Following on from the comment on Printer - is it actually useful to have names for CycleBreaker nodes at all or could they just be handled like Dummy nodes, where they're just templated on the "to" type, e.g. `CycleBreaker<MyType>`